src/Illuminate/Http/Resources/Json/Resource.php

```diff
@@ -91,15 +91,13 @@ public function resolve($request = null)
             $request = $request ?: Container::getInstance()->make('request')
         );
 
-        if (is_array($data)) {
-            $data = $data;
-        } elseif ($data instanceof Arrayable || $data instanceof Collection) {
-            $data = $data->toArray();
+        if ($data instanceof Arrayable) {
+            $data = $this->filter($data->toArray());
         } elseif ($data instanceof JsonSerializable) {
-            $data = $data->jsonSerialize();
+            $data = $this->filter($data->jsonSerialize());
```

Comment:
Why are you calling `filter` here & above?

Reply:
Oops...